bika/lims/browser/js/coffee/bika.lims.analysisservice.coffee

```diff
@@ -1,726 +1,1326 @@
+
+    options =
+      url: @get_portal_url() + "/@@API/read"
+      data:
+        catalog_name: "bika_setup_catalog"
+        page_size: 0
+        portal_type: "Instrument"
+
+    @ajax_submit options
+    .done (data) ->
+      if not data.objects
+        # resolve with an empty array
+        return deferred.resolveWith this, [[]]
+      # resolve with data objects
+      return deferred.resolveWith this, [data.objects]
+
+    return deferred.promise()
+
+
+  load_available_methods: =>
+    ###*
+     * Load all available and valid instruments
+     *
+     * @returns {Deferred} Array of all available Method objects
+    ###
+    deferred = $.Deferred()
+
+    options =
+      url: @get_portal_url() + "/@@API/read"
+      data:
+        catalog_name: "bika_setup_catalog"
+        page_size: 0
+        portal_type: "Method"
+        inactive_state: "active"
+        sort_on: "sortable_title"
+
+    @ajax_submit options
+    .done (data) ->
+      if not data.objects
+        # resolve with an empty array
+        return deferred.resolveWith this, [[]]
+      return deferred.resolveWith this, [data.objects]
+
+    return deferred.promise()
+
+
+  load_instrument_methods: (instrument_uid) =>
+    ###*
+     * Load assigned methods of the instrument
+     *
+     * @param {string} instrument_uid
+     *    The UID of the instrument
+     * @returns {Deferred} Array of Method objects
+    ###
+    deferred = $.Deferred()
+
+    # return immediately if we do not have a valid UID
+    if not @is_uid instrument_uid
+      deferred.resolveWith this, [[]]
+      return deferred.promise()
+
+    options =
+      url: @get_portal_url() + "/get_instrument_methods"
+      data:
+        uid: instrument_uid
+    @ajax_submit options
+    .done (data) ->
+      # {instrument: "51ebff9bf1314d00a7731b2f765dac37", methods: Array(3), title: "My Instrument"}
+      # where `methods` is an array of {uid: "3a85b7bc0430496ba7d0a6bcb9cdc5d5", title: "My Method"}
+      if not data.methods
+        # resolve with an empty array
+        deferred.resolveWith this, [[]]
+      deferred.resolveWith this, [data.methods]
+
+    return deferred.promise()
+
+
+  load_method_calculation: (method_uid) =>
+    ###*
+     * Load assigned calculation of the given method UID
+     *
+     * @param {string} method_uid
+     *    The UID of the method
+     * @returns {Deferred} Calculation object
+    ###
+    deferred = $.Deferred()
+
+    # return immediately if we do not have a valid UID
+    if not @is_uid method_uid
+      deferred.resolveWith this, [{}]
+      return deferred.promise()
+
+    options =
+      url: @get_portal_url() + "/get_method_calculation"
+      data:
+        uid: method_uid
+
+    @ajax_submit options
+    .done (data) ->
+      # /get_method_calculation returns just this structure:
+      # {uid: "488400e9f5e24a4cbd214056e6b5e2aa", title: "My Calculation"}
+      if not @is_uid data.uid
+        return deferred.resolveWith this, [{}]
+
```

Comment:
This "Instrument assignment is not required"/"Manual entry of results" thing is totally weird. It shouldn't rely on the assignment/unassignment of methods, rather with instruments (if no instrument assigned, then there is no choice, the user *must* set the result manually)... but in turn, Method content type has the option "Manual entry" True/False, at the same time that can have none or multiple instruments associated.... so well, kind of an hydra we'll have to deal with in future.